### ISSUE TITLE
Feature-complete test drive

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/geometric-optics#328]
_Geometric Optics_ is now feature-complete, and the design team is encourage to test drive it with high priority.  Please create new GitHub issues for any problems or change requests.   Unassign yourself when you have completed your test drive.

If there's anyone else who should be looking at the sim, please assign this issue to them.

- [x] @ariel-phet 
- [x] @arouinfar 
- [x] @kathy-phet 
- [x] @sarchang  (Optional)
- [x] @veillette  (Optional) 
- [x] @pixelzoom

**Comment by pixelzoom** [comment]
If there's anyone else who should be looking at the sim, please assign this issue to them.

**Comment by ariel-phet** [comment]
Test drive complete, overall sim is looking really good.

**Comment by pixelzoom** [comment]
Thanks @ariel-phet.

**Comment by pixelzoom** [comment]
@arouinfar @kathy-phet what is your status on this issue?

**Comment by kathy-phet** [comment]
I spent about 30 minutes working yesterday on review, but haven't quite finished. I'll try more today.

**Comment by pixelzoom** [comment]
3/3/2022 design meeting:

@arouinfar and @kathy-phet have both completed the test drive.

We went over @kathy-phet's feedback and created several issues (see above).

I'll invite @veillette and @sarchang to test-drive as soon as I fix #354 and create a dev version.

**Comment by pixelzoom** [comment]
@veillette and @sarchang If you have the time, we'd like to invite you to test-drive Geometric Optics, as we approach the finish line.  Please use [1.1.0-dev.32](https://phet-dev.colorado.edu/html/geometric-optics/1.1.0-dev.32/phet/geometric-optics_en_phet.html), and ideally complete this by 3/11/2022.  Be sure to check out the new "flat" mirror, and the additional features available via the PhET > Options menu.

**Comment by veillette** [comment]
I took the simulation for a ride. The simulation has come a long since  last August. I am particularly impressed with the arrow mode. Kudos to @pixelzoom because there must have been a fairly large internal restructure of the code base to support this representation. 

There are many small but significant enhancements.  The dashed virtual rays really emphasized the fact that they are virtual.
The number of rays (in many Mode) increased as the object is held further away from the optic, ensuring that there are always rays that passed through the optics.  
I appreciate the addition of the 2F points, and I support the removal of the guides for the Lens screen.

I noticed that in the size of the arrowhead does not rescale. I presume you have discussed it at length in design meetings, and you came to this decision for some good reasons. 

Well done.

**Comment by pixelzoom** [comment]
Thank you @veillette.  There were indeed a lot of changes to be made, but I had a good code base to build on. And it's a testament to you and @sarchang that I was able to understand the code and make major changes without having to consult you very often.  So "well done" to you too!

Re:
> I noticed that in the size of the arrowhead does not rescale. I presume you have discussed it at length in design meetings, and you came to this decision for some good reasons.

Yes, this was discussed quite a bit.  There's not a lot of backstory recorded in the issue, but I noted the decisions in https://github.com/phetsims/geometric-optics/issues/228#issuecomment-1039672404.  More backstory: My first implementation scaled all aspects of the arrows, as is done with images.  But the consensus was that the arrow is a representation where we should only be scaling the magnitude.  The head and tail dimensions therefore do not scale, while the magitude does.

**Comment by sarchang** [comment]
Thanks for including me in this! Overall, it looks much improved; thanks for all the work you put in to clean up and add to what we had started.
1. I was a bit confused about the projector mode situation: when I selected the projector mode, I only saw a couple colors change. Wasn’t sure if the font size of the labels were supposed to get larger as well. I also saw some discussion in the Colors #319 Github issue about whether or not to keep the projector mode. I do like the white background in terms of practicality (printing on paper) and visibility. 
2. The arrow representation is a really valuable addition.
3. With the y-axis constraint on the object, the most zoomed out setting feels a little weird since you can’t move the object very much in comparison to the size of the play area, but it probably makes sense because we usually don’t want to move the object too far from the optic anyway.
4. Under PhET > Options, it says “Show “2F Points” checkbox”. Not sure if this is just supposed to say “Show “2F Points”“?
These are my thoughts for now; might come back to this later if I have some time tonight. Thanks for all the great work!

**Comment by pixelzoom** [comment]
Thanks @sarchang!  To address your feedback:

> 1. I was a bit confused about the projector mode situation: ...

We've moved away from the original "dark background" color scheme, so "Projector mode" is now somewhat obsolete.  There was a proposal to make it go away entirely in https://github.com/phetsims/geometric-optics/issues/275. @arouinfar is now going to address that as a general revision of the color profile(s) in https://github.com/phetsims/geometric-optics/issues/319. 

> 2. The arrow representation is a really valuable addition.

Glad you like it. @ariel-phet was the primary advocate for adding this representation, since it seems to be the most typical representation used in courses.

> 3. With the y-axis constraint on the object, the most zoomed out setting feels a little weird ...

When the object gets close to the Object, but is far _above_ the optic, we ran into cases where no "Many" rays reached the optic, and thus no Image was formed.  See for example the screenshot below from 1.1.0-dev.17. To avoid these confusing cases, we decided to limit dragging of the Object to 100 cm above or below the optical axis. More details in https://github.com/phetsims/geometric-optics/issues/289#issuecomment-1018019517.

<img width="999" alt="screenshot_1600" src="https://user-images.githubusercontent.com/3046552/157962192-b7db5fd5-da19-48fc-bf8b-29a38411d844.png">


> 4. Under PhET > Options, it says “Show “2F Points” checkbox”. Not sure if this is just supposed to say “Show “2F Points”“?

Actually, this checkbox does not show the "2F Points".  It adds an additional "2F Points" checkbox to the control panel.  Hence the name "Show “2F Points” checkbox”.  Confusing, I agree, but friendlier than the more-accurate "Enable 2F Points feature".
